rfcs/000-ml-feature-discovery.md

```diff
@@ -0,0 +1,183 @@
+    - ex: test_feature_group_name/test_feature_name/1.2.0
+- Availability/sources: list of places where feature is available
+- Name: the name of the feature
+- Version: the version of the feature
+- Data type
+- Description: optional, a description of the feature
+- Owners: individual and team emails of owners
+- Feature group: a way of grouping features, can be defined differently in each company, in this case we assume this will be the name of the pipeline that produced the features
+- Entity: a way of grouping features semantically by using nouns
+- Owner tags: programmatically generated and not editable tags, useful to group features base don the specific engineering team using them
+- User tags: normal amundsen tags
+- Contains PII: weather a feature contains PII or not, will be a badge in the badges list
+- Date range: what time frame is the data in the feature for
+- Partition column: optional column associated to the feature to indicate partition
+
+#### Operational Metadata
+Metadata to facilitate the use of the feature by other engineers.
+- Code snippet/SQL: snippet of code that generated the feature and external URL to more details
+- Last updated: timestamp for when the feature was most recently updated
+- Usage: number of accesses to the feature, usage in models out of scope for MVP since we are not indexing models yet
+- Upstream resources: upstream lineage in the form of tables, could be expanded to anything else in the future (column, service, pipeline)
+#### Data Pattern Metadata
+Metadata that describes qualities of the data in this feature.
+- Data sample: a sample of the data which the feature contains shown as a table
+- Statistics: any stats relevant to the feature data
+
+
+## UI/UX-level Explanation
+_**Note:** The pink text is only visible in the mocks to call out design details and changes._
+### Home Page
+![Table Lineage Preview](../assets/036/home-page.png)
+##### Expanded Search Bar
+![Table Lineage Preview](../assets/036/home-page-expanded.png)
+### Advanced Search Page
+![Table Lineage Preview](../assets/036/adv-search.png)
+
+### Feature Details Page
+##### Sample Tab
+![Table Lineage Preview](../assets/036/sample-tab.png)
+##### Statistics Tab
+![Table Lineage Preview](../assets/036/stats-tab.png)
+##### Upstream Tab
+![Table Lineage Preview](../assets/036/upstream-tab.png)
+##### Generation Code Tab
+![Table Lineage Preview](../assets/036/generation-tab.png)
+
+## Reference-level Explanation (aka Technical Details)
+
+### Amundsen Databuilder
+The databuilder library will be expanded to include a feature Model that includes the following:
+#### Nodes
+- Feature node:
+    - Key
+    - Name
+    - Version
+    - Feature group
+    - Entity
+    - Data type
+    - Last updated
+#### Relationships
+Here we show the relationships between feature and other nodes, as well as the metadata that this represents: 
```

Comment:
We are currently doing this by having an upstream tab that relates to tables rather than specific columns, in the future we can expand this to be both or even integrate upstream services rather than datasets.